### PR TITLE
Switch to Google tag manager

### PR DESCRIPTION
**Pull request by aileenrose**
See https://github.com/sourcegraph/sourcegraph/issues/12989

### CODE REVIEW

File: doc/_resources/templates/root.html [hunk 1]
```diff
@@ -8,9 +8,20 @@
     <link rel="stylesheet" href="{{asset "search.css"}}?19" />
     <link rel="stylesheet" href="{{asset "content.css"}}?19" />
     <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no, viewport-fit=cover" />
+    <!-- Google Tag Manager -->
```
Comment (dadlerj) [inline doc/_resources/templates/root.html:11]: We don't ship a docs page to customers, right? It just links back to our docs.sourcegraph.com? I want to be sure that there's zero chance of GTM being enabled on a customer deployment.

File: enterprise/dev/Procfile [hunk 2]
```diff
@@ -5,7 +5,7 @@ searcher: searcher
 replacer: replacer
 symbols: symbols
 github-proxy: github-proxy
-frontend: env CONFIGURATION_MODE=server SITE_CONFIG_ESCAPE_HATCH_PATH=$HOME/.sourcegraph/site-config.json frontend
+frontend: env SOURCEGRAPHDOTCOM_MODE=TRUE CONFIGURATION_MODE=server SITE_CONFIG_ESCAPE_HATCH_PATH=$HOME/.sourcegraph/site-config.json frontend
```
Comment (daxmc99) [inline enterprise/dev/Procfile:8]: I'm not sure if we should enable `SOURCEGRAPHDOTCOM_MODE=TRUE` for all of our developers. This should likely only be done as needed. Maybe adding an option to the `dev/start.sh` script would be preferable.

File: web/src/search/input/SearchPage.tsx [hunk 3]
```diff
@@ -90,7 +90,7 @@ export const SearchPage: React.FunctionComponent<Props> = props => {
             [codeInsightsEnabled, props.extensionsController.services.view]
         )
     )
-
+    console.log('###', props)
```
Comment (daxmc99) [inline web/src/search/input/SearchPage.tsx:93]: Debug print?

File: web/src/repogroups/Stanford.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                    <span className="repogroup-page__keyword-text">repo:</span>/RedHeadedStranger99/{' '}
+                </>
+            ),
+            patternType: SearchPatternType.literal,
+            rawQuery: 'repo:/RedHeadedStranger99/',
+        },
+    ],
+    homepageDescription: 'Explore Stanford open-source code.',
+    homepageIcon:
+        'https://upload.wikimedia.org/wikipedia/en/thumb/b/b7/Stanford_University_seal_2003.svg/1200px-Stanford_University_seal_2003.svg.png',
```
Comment (daxmc99) [inline web/src/repogroups/Stanford.tsx:76]: This looks like [hotlinking](https://codeburst.io/hotlinking-fef83c9a7518)

We should make sure we have permission to use this icon and host it ourselves

**Comment by daxmc99** [comment]
@aileenrose It looks like parts of this PR add a Stanford repogroup to the homepage. Is there a separate issue you could link for that?

File: dev/global-settings.json [hunk 5]
```diff
@@ -1,4 +1,6 @@
 {
   // This is set from an environment variable
-  "experimentalFeatures": {}
+  "experimentalFeatures": {
+    "showRepogroupHomepage": true
```
Comment (unknwon) [inline dev/global-settings.json:4]: If the purpose is to enable in dev environment, should generally be added here: https://github.com/sourcegraph/dev-private/blob/master/enterprise/dev/site-config.json#L3

File: enterprise/dev/Procfile (hunk 2 above)
Comment (unknwon) [reply enterprise/dev/Procfile:8]: `SOURCEGRAPHDOTCOM_MODE=TRUE` has many assumptions about set up I think, better to have a more specific option if required.

**Comment by aileenrose** [comment]
> @aileenrose It looks like parts of this PR add a Stanford repogroup to the homepage. Is there a separate issue you could link for that?

I am not sure how I picked up @RedHeadedStranger99  commits which been merged in this PR https://github.com/sourcegraph/sourcegraph/pull/12791
I'll merge with main again.

File: doc/_resources/templates/root.html (hunk 1 above)
Comment (aileenrose) [reply doc/_resources/templates/root.html:11]: I verified that help links in the local installation of 3.18 go to docs.sourcegraph.com.

**Comment by aileenrose** [comment]
@daxmc99 logo addressed in https://github.com/sourcegraph/sourcegraph/pull/13257

File: web/src/search/input/SearchPage.tsx (hunk 3 above)
Comment (daxmc99) [inline web/src/search/input/SearchPage.tsx:93]:
```suggestion
```

File: enterprise/dev/Procfile (hunk 2 above)
Comment (daxmc99) [reply enterprise/dev/Procfile:7]:
```suggestion
frontend: env CONFIGURATION_MODE=server SITE_CONFIG_ESCAPE_HATCH_PATH=$HOME/.sourcegraph/site-config.json frontend
```

**Pull request merged**